Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -442,13 +446,27 @@ def process_alert(self, alert, alert_type):
             self._logger.debug("LibtorrentDownloadImpl: alert %s with message %s", alert_type, alert)
 
         alert_types = ('tracker_reply_alert', 'tracker_error_alert', 'tracker_warning_alert', 'metadata_received_alert',
-                       'file_renamed_alert', 'performance_alert', 'torrent_checked_alert', 'torrent_finished_alert')
+                       'file_renamed_alert', 'performance_alert', 'torrent_checked_alert', 'torrent_finished_alert',
+                       'save_resume_data_alert')
 
         if alert_type in alert_types:
             getattr(self, 'on_' + alert_type)(alert)
         else:
             self.update_lt_stats()
 
+    def on_save_resume_data_alert(self, alert):
+        """
+        alert to handle save_resume_data_alert
+        it will assign store resume data in a attribute,
+            and called save_download_pstate in LaunchManyCore
+        """
+        self.resume_data = alert.resume_data
+
+        # at this point, Deferred object should call save_download
+        if self.saveresume_called_defer:
+            self.saveresume_called_defer.callback(alert.resume_data)
+            self.saveresume_called_defer = None
```

Comment:
Apparently this is not an issue. Made a small example and this works: https://gist.github.com/lfdversluis/68af55e43bfc167f6617f5debd9fbad2